torch/csrc/jit/passes/onnx/fixup_onnx_controlflow.cpp

```diff
@@ -258,25 +259,43 @@ bool IsUninitializedNode(Node* n) {
 
 // Infer shape and type of the uninitialized_output from the corresponding
 // output of the other subblock. prim::Uninitialized node is proven to be
-// unused. So replace this node with a constant of the inferred shape and type.
+// unused. So replace this node with a constant(TensorType) or
+// Sequence(ListType) of the inferred shape and type.
 void InferShapeTypeForUninitializedOutput(
     Graph* graph,
     Block* block,
     Value* uninitialized_output,
     Value* other_output) {
-  auto output_type = other_output->type()->expect<TensorType>();
-  auto elem_type = at::initialTensorOptions().dtype(output_type->scalarType());
-  Node* const_node = graph->create(::c10::onnx::Constant, 1);
-
-  if (output_type->sizes().concrete_sizes().has_value()) {
-    auto size = output_type->sizes().concrete_sizes().value();
-    const_node->t_(attr::value, at::zeros(size, elem_type));
-    const_node->output()->setType(other_output->type());
-    const_node->output()->copyMetadata(other_output);
-  } else {
-    const_node->t_(attr::value, at::zeros({}, elem_type));
-    const_node->output()->setType(
-        TensorType::create(*(output_type->scalarType()), at::kCPU, {}, {}));
+  Node* const_node = nullptr;
+  if (auto output_type = other_output->type()->cast<TensorType>()) {
+    auto elem_type =
+        at::initialTensorOptions().dtype(output_type->scalarType());
+    const_node = graph->create(::c10::onnx::Constant, 1);
+
+    if (output_type->sizes().concrete_sizes().has_value()) {
+      auto size = output_type->sizes().concrete_sizes().value();
+      const_node->t_(attr::value, at::zeros(size, elem_type));
+      const_node->output()->setType(other_output->type());
+    } else {
+      const_node->t_(attr::value, at::zeros({}, elem_type));
+      const_node->output()->setType(
+          TensorType::create(*(output_type->scalarType()), at::kCPU, {}, {}));
+    }
+  } else if (auto output_type = other_output->type()->cast<ListType>()) {
+    TypePtr elem = output_type->getElementType();
+    const_node = graph->create(::c10::onnx::SequenceEmpty, 1);
+    if (elem->cast<TensorType>() &&
+        elem->cast<TensorType>()->scalarType().has_value()) {
+      auto scalar_type = elem->cast<TensorType>()->scalarType().value();
+      auto onnx_type = ATenTypeToOnnxType(scalar_type);
+      const_node->i_(attr::dtype, onnx_type);
+      const_node->output()->setType(other_output->type());
+    } else {
```

Comment:
I think we can also handle scalar type lists like (int[], float[], etc.) here.
Maybe like:
else if (elem->cast<IntType>()) {
      auto scalar_type = at::kLong;
      auto onnx_type = ATenTypeToOnnxType(scalar_type);
      const_node->i_(attr::dtype, onnx_type);
      const_node->output()->setType(other_output->type());
}

Reply:
Good point! Will add this part and tests.
By the way, Do you think we also need to add scalar type like(int, float etc..) in output_type?
https://github.com/pytorch/pytorch/blob/c3eb80300af37aa41b9e7ff2d3ff2220393db6b7/torch/csrc/jit/passes/onnx/fixup_onnx_controlflow.cpp#L284

Reply:
It seems the ONNX node type must be Tensor type. Even we return like the `intList`, it will also return the element tensor type. So we may not need to handle those type.
I have added this in test case.